plugins/polio/js/src/forms/DetectionForm.js

```diff
@@ -15,15 +15,62 @@ export const detectionFormFields = [
 
 export const DetectionForm = () => {
     const { formatMessage } = useSafeIntl();
+    const { values, setFieldValue } = useFormikContext();
+    const updateOnset = useCallback(
+        (fieldName, date) => {
+            if (date && !values.cvdpv2_notified_at && !values.pv_notified_at) {
+                setFieldValue('detection_status', 'ONGOING');
+            }
+            if (!date && !values.cvdpv2_notified_at && !values.pv_notified_at) {
+                setFieldValue('detection_status', 'PENDING');
+            }
+        },
+        [setFieldValue, values.cvdpv2_notified_at, values.pv_notified_at],
+    );
+    const updatePVNotification = useCallback(
+        (fieldName, date) => {
+            if (date && !values.cvdpv2_notified_at) {
+                setFieldValue('detection_status', 'ONGOING');
+            }
+            if (!date && !values.cvdpv2_notified_at && !values.onset_at) {
+                setFieldValue('detection_status', 'PENDING');
+            }
+        },
+        [setFieldValue, values.cvdpv2_notified_at, values.onset_at],
+    );
+    const updateVirusNotification = useCallback(
+        (fieldName, date) => {
+            if (date) {
+                setFieldValue('detection_status', 'FINISHED');
+            }
+            if (!date && !values.onset_at && !values.pv_notified_at) {
+                setFieldValue('detection_status', 'PENDING');
+            }
+            if (!date && (values.onset_at || values.pv_notified_at)) {
+                setFieldValue('detection_status', 'ONGOING');
+            }
+        },
+        [setFieldValue, values.onset_at, values.pv_notified_at],
+    );
     return (
         <>
             <Grid container spacing={2}>
                 <Grid container direction="row" item spacing={2}>
                     <Grid xs={12} md={6} item>
-                        <Field
-                            name="detection_status"
-                            component={StatusField}
-                        />
+                        <Box mb={2} px={2} py={2}>
+                            <Typography variant="button">
+                                {`${formatMessage(
+                                    MESSAGES.status,
+                                )}: ${formatMessage(
+                                    MESSAGES[
+                                        values.detection_status.toLowerCase()
```

Comment:
be careful with this if MESSAGES doesn't contain the status, it will fail. Maybe double check this before and display the key if missing